lib/req/steps.ex

```diff
@@ -2059,6 +2069,232 @@ defmodule Req.Steps do
     {request, response}
   end
 
+  @doc """
+  Handles HTTP Digest authentication when auth is provided as `{:digest, userinfo}`
+
+  ## Examples
+
+      iex> Req.get!("https://httpbin.org/digest-auth/auth/user/pass", auth: {:digest, "user:pass"})
+      {:ok, %Req.Response{status: 200, body: %{"authenticated" => true, "user" => "user"}}}
+  """
+  @doc step: :response
+  def handle_http_digest({request, %Req.Response{status: 401} = response}) do
+    case request.options[:auth] do
+      {:digest, userinfo} when is_binary(userinfo) ->
+        [username, password] = String.split(userinfo, ":", parts: 2)
+
+        handle_challenge_reply(request, response,
+          username: username,
+          password: password
+        )
+
+      _ ->
+        {request, response}
+    end
+  end
+
+  def handle_http_digest(other), do: other
+
+  defp handle_challenge_reply(request, response, opts) do
+    username = Keyword.get(opts, :username, "")
+    password = Keyword.get(opts, :password, "")
+    count = Keyword.get(opts, :count, 1)
```

Comment:
`:count` seems to be always 1? The only call site I found is:

```elixir
        handle_challenge_reply(request, response,
          username: username,
          password: password
        )
```

which does not set :count.

Reply:
Ah, I suppose the response for the second request would not have www-authenticate header anymore so we'd have to store it for later and add API to grab it, perhaps something like `Req.Response.get_auth_digest_data(resp)` but yeah, not loving it 😅

Reply:
Yes, your understanding is correct about the use case here (reducing # of requests)

I do like the idea of a helper function to retrieve the data and passing it back in on subsequent requests, since it gives the user full control and doesn't require maintaining an internal state. It could retrieve the data needed for subsequent requests in regard to the current nonce and count (depending on whether a `nextnonce` was returned)

I agree, let's wait for now if that sounds good to you, if we need this later I can take a shot at implementing it :)